src/problem/solve_problem.jl

```diff
@@ -126,20 +135,24 @@ end
         k::Int=1,
         disturbance_clusters::Set{Int64}=Set{Int64}()
         seed::Union{Nothing,Int64}=nothing,
-        rng::AbstractRNG=Random.GLOBAL_RNG
+        rng::AbstractRNG=Random.GLOBAL_RNG,
+        waypoint_optim_flag::Bool=true,
     )::MSTSolution
 
 Generate a solution to the problem for:
 - the mothership by using the nearest neighbour heuristic to generate an initial solution,
     and then improving using the 2-opt heuristic, and
 -  for tenders using the sequential nearest neighbour heuristic.
+Optimize solution using simulated annealing.
+Optionally, optimize waypoints using gradient descent.
```

Comment:
Suggest not hardcoding names of methods because these can change and will inevitably lead to confusion in the future.